Orange/widgets/data/owfeaturestatistics.py

```diff
@@ -290,52 +290,45 @@ def __mode(x, *args, **kwargs):
-
-        Returns
-        -------
-        Union[Tuple[List[str], np.ndarray], np.ndarray]
-
-        """
-        if self.table is None:
-            return np.atleast_2d([])
+    def get_statistics_table(self):
+        """Get the numeric computed statistics in a single matrix."""
+        if self.table is None or not self.rowCount():
+            return None
 
-        # If a list of variables is given, select only corresponding stats
-        # variables can be a list or array, pylint: disable=len-as-condition
-        if variables is not None and len(variables) != 0:
-            indices = [self.domain.index(var) for var in variables]
+        # don't match TimeVariable, pylint: disable=unidiomatic-typecheck
+        contivars = [type(var) is ContinuousVariable for var in self.variables]
+        if any(contivars):
+            def c(column):
+                return np.choose(contivars, [np.nan, column])
+
+            x = np.vstack((
+                c(self._center), c(self._median), self._dispersion,
+                c(self._min), c(self._max), self._missing,
+            )).T
+            attrs = [ContinuousVariable(column.name) for column in (
+                self.Columns.CENTER, self.Columns.MEDIAN,
+                self.Columns.DISPERSION,
+                self.Columns.MIN, self.Columns.MAX, self.Columns.MISSING)]
         else:
-            indices = ...
-
-        matrix = np.vstack((
-            self._center[indices], self._median[indices],
-            self._dispersion[indices],
-            self._min[indices], self._max[indices], self._missing[indices],
-        )).T
-
-        # Return string labels for the returned matrix columns e.g. 'Mean',
-        # 'Dispersion' if requested
-        if return_labels:
-            labels = [self.Columns.CENTER.name, self.Columns.MEDIAN.name,
-                      self.Columns.DISPERSION.name,
-                      self.Columns.MIN.name, self.Columns.MAX.name,
-                      self.Columns.MISSING.name]
-            return labels, matrix
+            x = np.vstack((self._dispersion, self._missing)).T
+            attrs = [ContinuousVariable(name)
+                     for name in ("Entropy", self.Columns.MISSING.name)]
+
+        names = [var.name for var in self.variables]
+        if any(isinstance(var, DiscreteVariable) for var in self.variables):
+            majorities = [
+                var.str_val(val) if isinstance(var, DiscreteVariable) else ""
+                for var, val in zip(self.variables, self._median)]
+            metas = np.vstack((names, majorities)).T
+            meta_attrs = [StringVariable('Feature'), StringVariable('Majority')]
+        else:
+            metas = np.atleast_2d(names).T
+            meta_attrs = [StringVariable('Feature')]
 
-        return matrix
+        domain = Domain(attributes=attrs, metas=meta_attrs)
+        statistics = Table.from_numpy(domain, x, metas=metas)
+        statistics.name = '{self.data.name} (Feature Statistics)'
```

Comment:
It's better to:
1) put an f before this string
2) leave this line in `OWFeatureStatistics.commit`, where self.data.name actually exists

:bug: :bug: :bug: 
If you move it outside of this function, don't forget that this can return None in which case you should not set the name...

Reply:
I just noticed, that you can probably just change self.data to self.table and leave the line where it is.